src/main/java/at/hannibal2/skyhanni/features/mining/GemstoneMoneyPerHour.kt

```diff
@@ -0,0 +1,167 @@
+        "mining.pristine",
+        "§d§lPRISTINE! §r§fYou found §r§a. Flawed (?<gemstone>\\w+) Gemstone §r§8x(?<amount>\\d+)§r§f!"
+    )
+
+    private val config get() = SkyHanniMod.feature.mining.gemstoneMoneyPerHour
+
+    private var display: List<Renderable> = listOf()
+    private var start = SimpleTimeMark.farPast()
+    private var lastMined = SimpleTimeMark.farPast()
+    private var coins = 0
+    private var lastGemstone: String = ""
+    private var uptime: Duration = 0.seconds
+    private var paused: Boolean = false
+
+    @HandleEvent(onlyOnSkyblock = true)
+    fun onChat(event: SkyHanniChatEvent) {
+        if (!isEnabled()) return
+        pristineMessagePattern.matchMatcher(event.message) {
+            if (start.isFarPast()) start = SimpleTimeMark.now()
+            if (paused) paused = false
+            lastMined = SimpleTimeMark.now()
+            val gemstone = group("gemstone")
+            lastGemstone = gemstone
+            val configGemstonePrice = getPrice(convertToInternalName(lastGemstone))
+            val delta = group("amount").toDouble() * getFraction() * configGemstonePrice
+            coins += delta.toInt()
+        }
+    }
+
+    private fun convertToInternalName(name: String): NeuInternalName {
+        return "${config.gemstoneType.displayName}_${name}_GEM".toInternalName()
+    }
+
+    private fun getPrice(gemstone: NeuInternalName): Double {
+        return if (config.forceNPC) gemstone.getNpcPrice()
+        else maxOf(gemstone.getNpcPrice(), gemstone.getPrice())
+    }
+
+    private fun getFraction(): Double {
+        return (80.0).pow(2 - toNum(config.gemstoneType))
+    }
+
+    private fun toNum(type: GemstoneMoneyPerHourConfig.GemstoneType): Int {
+        return when (type) {
+            GemstoneMoneyPerHourConfig.GemstoneType.ROUGH -> 1
+            GemstoneMoneyPerHourConfig.GemstoneType.FLAWED -> 2
+            GemstoneMoneyPerHourConfig.GemstoneType.FINE -> 3
+            GemstoneMoneyPerHourConfig.GemstoneType.FLAWLESS -> 4
+        }
+    }
+
+    private fun updateDisplay() {
+        display = createDisplay()
+    }
+
+    private fun createDisplay() = buildList {
+        if (start.isFarPast()) return@buildList
+        if (lastGemstone.isEmpty()) return@buildList
+        val moneyPerHour = coins / maxOf(uptime.inPartialSeconds, 1.0) * 3600
+        val internalName = convertToInternalName(lastGemstone)
+        val gemstoneName = internalName.itemNameWithoutColor
+        val gemstonePrice = getPrice(internalName).toInt()
+        val pausedText = if (paused) " §c(PAUSED)"
+        else ""
+
+        add(Renderable.string("§d§lGemstone Money per Hour"))
+        add(Renderable.string("§a($gemstoneName @ §b${gemstonePrice.shortFormat()})"))
+        add(Renderable.string("§a$/hr: §b${moneyPerHour.toInt().shortFormat()}"))
+        add(Renderable.string("§a$ made: §b${coins.shortFormat()}"))
+        add(Renderable.string("§aUptime: §b${uptime.format()}$pausedText"))
```

Comment:
I personally prefer using "Coins" over the dollar sign because that not how coins are represented in game

Reply:
I agree